File: scripts/export-icon.py
#!/usr/bin/env python3
import cairosvg
from PIL import Image
import io
import os

ASSETS = '/Users/chris.litster/Library/CloudStorage/OneDrive-SumeruEquityPartners/Desktop/AskFuji/assets'
SVG_PATH = os.path.join(ASSETS, 'icon.svg')

def svg_to_png(size, out_path):
    png = cairosvg.svg2png(url=SVG_PATH, output_width=size, output_height=size)
    img = Image.open(io.BytesIO(png))
    img.save(out_path)
    print(f'✓ {out_path} ({size}x{size})')

# Main icon (1024x1024)
svg_to_png(1024, os.path.join(ASSETS, 'icon.png'))

# Splash icon (centered on white, 288x288 icon on 1284x2778 canvas)
splash_bg = Image.new('RGBA', (1284, 2778), (26, 58, 42, 255))
icon_png = cairosvg.svg2png(url=SVG_PATH, output_width=288, output_height=288)
icon_img = Image.open(io.BytesIO(icon_png))
x = (1284 - 288) // 2
y = (2778 - 288) // 2
splash_bg.paste(icon_img, (x, y), icon_img)
splash_bg.save(os.path.join(ASSETS, 'splash-icon.png'))
print(f'✓ splash-icon.png')

# Adaptive icon foreground (108dp = 432px with safe zone)
svg_to_png(432, os.path.join(ASSETS, 'adaptive-icon.png'))

print('\nAll icons exported.')

File: scripts/export-icon.mjs
import sharp from 'sharp';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const ASSETS = path.join(__dirname, '../assets');
const SVG_PATH = path.join(ASSETS, 'icon.svg');
const svgBuffer = fs.readFileSync(SVG_PATH);

async function exportIcons() {
  // Main icon 1024x1024
  await sharp(svgBuffer).resize(1024, 1024).png().toFile(path.join(ASSETS, 'icon.png'));
  console.log('✓ icon.png (1024x1024)');

  // Adaptive icon 432x432
  await sharp(svgBuffer).resize(432, 432).png().toFile(path.join(ASSETS, 'adaptive-icon.png'));
  console.log('✓ adaptive-icon.png (432x432)');

  // Splash icon — icon centered on green background
  const iconBuffer = await sharp(svgBuffer).resize(288, 288).png().toBuffer();
  await sharp({
    create: { width: 1284, height: 2778, channels: 4, background: { r: 26, g: 58, b: 42, alpha: 1 } }
  })
    .composite([{ input: iconBuffer, left: 498, top: 1245 }])
    .png()
    .toFile(path.join(ASSETS, 'splash-icon.png'));
  console.log('✓ splash-icon.png (1284x2778)');

  console.log('\nAll icons exported.');
}

exportIcons().catch(console.error);
